### PR TITLE
DM suggester on acceptance / denial

### PR DESCRIPTION
**Pull request by hanatic**
See https://discord.com/channels/655109296400367618/709048920856330240/864434875858944020

### CODE REVIEW

**Comment by Milo123459** [comment]
Please merge #235 with this PR. @hanatic

**Comment by hanatic** [comment]
TSC seems to be failing despite it running fine locally. Merging #235 as we speak.

File: src/commands/Suggestion Commands/AcceptCommand.ts [hunk 1]
```diff
@@ -64,6 +64,16 @@ export const run: RunFunction = async (client, message, args) => {
 			color: 'GREEN',
 		})
 	);
+	
+	const author: User = client.users.cache.get(Suggestion.User);
```
Comment (Milo123459) [inline src/commands/Suggestion Commands/AcceptCommand.ts:68]:
```suggestion
const author: User = client.users.cache.get((Suggestion as Anything).User);
```

File: src/commands/Suggestion Commands/AcceptCommand.ts [hunk 2]
```diff
@@ -64,6 +64,16 @@ export const run: RunFunction = async (client, message, args) => {
...
+	author.send(
+		new MessageEmbed({
+			name: 'Your suggestion was accepted!',
```
Comment (Milo123459) [inline src/commands/Suggestion Commands/AcceptCommand.ts:71]:
```suggestion
name: `Your suggestion was accepted in ${message.guild.name}`

File: src/commands/Suggestion Commands/AcceptCommand.ts [hunk 3]
```diff
@@ -64,6 +64,16 @@ export const run: RunFunction = async (client, message, args) => {
...
+			value: `A staff member accepted your suggestion: ${Suggestion.Content} `,
```
Comment (Milo123459) [inline src/commands/Suggestion Commands/AcceptCommand.ts:72]:
```suggestion
description: `A staff member accepted your suggestion: ${(Suggestion as Anything).Content}`
```

File: src/commands/Suggestion Commands/AcceptCommand.ts [hunk 4]
```diff
@@ -64,6 +64,16 @@ export const run: RunFunction = async (client, message, args) => {
 			color: 'GREEN',
 		})
 	);
+	
+	const author: User = client.users.cache.get(Suggestion.User);
+	author.send(
```
Comment (Milo123459) [inline src/commands/Suggestion Commands/AcceptCommand.ts:69]: We need to write a check to see if the author is in the cache. A simple `if(author)` would work.

Comment (hanatic) [reply src/commands/Suggestion Commands/AcceptCommand.ts]: Updated to add check.

**Comment by Milo123459** [comment]
Thanks. Does this apply on both files?

File: src/commands/Suggestion Commands/AcceptCommand.ts [hunk 5]
```diff
@@ -64,6 +64,20 @@ export const run: RunFunction = async (client, message, args) => {
 			color: 'GREEN',
 		})
 	);
+	
+	const author: User = client.users.cache.get(Suggestion.User);
```
Comment (Milo123459) [inline src/commands/Suggestion Commands/AcceptCommand.ts:68]:
```suggestion
const author: User = client.users.cache.get((Suggestion as Anything).User)
```

File: src/commands/Suggestion Commands/AcceptCommand.ts [hunk 6]
```diff
@@ -64,6 +64,20 @@ export const run: RunFunction = async (client, message, args) => {
 			color: 'GREEN',
 		})
 	);
+	
+const author: User = client.users.cache.get((Suggestion as Anything).User);
+	if (author) {
+		author.send(
+			new MessageEmbed({
+				name: 'Your suggestion was accepted!',
+				value: `A staff member accepted your suggestion: ${Suggestion.Content} `,
```
Comment (Milo123459) [inline src/commands/Suggestion Commands/AcceptCommand.ts:73]: Change `value` to `description`

File: src/commands/Suggestion Commands/AcceptCommand.ts [hunk 7]
```diff
@@ -64,6 +64,20 @@ export const run: RunFunction = async (client, message, args) => {
 			color: 'GREEN',
 		})
 	);
+	
+const author: User = client.users.cache.get((Suggestion as Anything).User);
+	if (author) {
+		author.send(
+			new MessageEmbed({
+				name: 'Your suggestion was accepted!',
```
Comment (Milo123459) [inline src/commands/Suggestion Commands/AcceptCommand.ts:72]: `name` -> `title`

File: src/commands/Suggestion Commands/AcceptCommand.ts (hunk 6 above)
Comment (Milo123459) [reply src/commands/Suggestion Commands/AcceptCommand.ts:73]: And type cast Suggestion: `(Suggestion as Anything).Content`

File: src/commands/Suggestion Commands/AcceptCommand.ts (hunk 7 above)
Comment (hanatic) [reply src/commands/Suggestion Commands/AcceptCommand.ts:72]: i'm such an idiot. :)

File: src/commands/Suggestion Commands/AcceptCommand.ts [hunk 8]
```diff
@@ -64,6 +64,20 @@ export const run: RunFunction = async (client, message, args) => {
...
+				value: `A staff member accepted your suggestion: ${Suggestion.Content} `,
+				color: 'GREEN'
+			})
+		);
+	} else {
+		console.log('An error occured: could not find suggestion author in database');
```
Comment (Milo123459) [inline src/commands/Suggestion Commands/AcceptCommand.ts:78]: We don't need to log anything or need the else. The only reason we check is to make sure we don't get any issues if the user isn't cached.

File: src/commands/Suggestion Commands/AcceptCommand.ts (hunk 7 above)
Comment (Milo123459) [reply src/commands/Suggestion Commands/AcceptCommand.ts]: Don't worry about it! :)

File: src/commands/Suggestion Commands/AcceptCommand.ts (hunk 8 above)
Comment (hanatic) [reply src/commands/Suggestion Commands/AcceptCommand.ts]: Removed :)

File: src/commands/Suggestion Commands/AcceptCommand.ts [hunk 9]
```diff
@@ -64,6 +64,18 @@ export const run: RunFunction = async (client, message, args) => {
 			color: 'GREEN',
 		})
 	);
+	
+const author: User = client.users.cache.get((Suggestion as Anything).User);
+	if (author) {
+		author.send(
+			new MessageEmbed({
```
Comment (Milo123459) [inline src/commands/Suggestion Commands/AcceptCommand.ts:71]: Can we use the `client.embed` function rather then constructing manually? It'd show the user that accepted it.

Comment (hanatic) [reply src/commands/Suggestion Commands/AcceptCommand.ts:71]: Implementing now.

Comment (Milo123459) [reply src/commands/Suggestion Commands/AcceptCommand.ts:71]: FYI: Usage is `client.embed(stuff inside messageembed, message)`

**Comment by hanatic** [comment]
Error in both commands:
```log
Error: src/commands/Suggestion Commands/AcceptCommand.ts(71,11): error TS2554: Expected 2 arguments, but got 1. 
Error: src/commands/Suggestion Commands/DenyCommand.ts(72,11): error TS2554: Expected 2 arguments, but got 1.
```

File: src/commands/Suggestion Commands/AcceptCommand.ts [hunk 10]
```diff
@@ -64,6 +64,18 @@ export const run: RunFunction = async (client, message, args) => {
...
 	);
+	
+const author: User = client.users.cache.get((Suggestion as Anything).User);
+	if (author) {
+		author.send(
+			client.embed({
+				title: 'Your suggestion was accepted!',
+				description: `A staff member accepted your suggestion: ${(Suggestion as Anything).Content} `,
+				color: 'GREEN'
+			})
```
Comment (Milo123459) [inline src/commands/Suggestion Commands/AcceptCommand.ts:75]: Just add a `, message` after the `}` to fix this error.

**Comment by Milo123459** [comment]
> Error in both commands:
> 
> ```
> Error: src/commands/Suggestion Commands/AcceptCommand.ts(71,11): error TS2554: Expected 2 arguments, but got 1. 
> Error: src/commands/Suggestion Commands/DenyCommand.ts(72,11): error TS2554: Expected 2 arguments, but got 1.
> ```

Yes, I made a review which says how to fix it.

**Comment by Milo123459** [comment]
Also, when you get the chance please run `yarn lint` to format code.

File: src/commands/Suggestion Commands/DenyCommand.ts [hunk 11]
```diff
@@ -64,6 +64,19 @@ export const run: RunFunction = async (client, message, args) => {
...
+	
+	const author: User = client.users.cache.get((Suggestion as Anything).User);
+
+	if (author) {
+		author.send(
+			client.embed({
+				title: `Your suggestion was denied in ${message.guild.name}`,
+				description: `A staff member denied your suggestion: ${(Suggestion as Anything).Content} `,
+				color: 'RED'
+			})
```
Comment (Milo123459) [inline src/commands/Suggestion Commands/DenyCommand.ts:76]: No, inside the `client.embed` function 😅

File: src/commands/Suggestion Commands/AcceptCommand.ts (hunk 9 above)
Comment (hanatic) [reply src/commands/Suggestion Commands/AcceptCommand.ts]: i am such. an. idiot.

Comment (Milo123459) [reply src/commands/Suggestion Commands/AcceptCommand.ts]: Don't worry about it. Seriously!

**Comment by Milo123459** [comment]
Please run `yarn lint` then commit the code so that the CI doesn't fail.

**Comment by hanatic** [comment]
okay :)

**Comment by Milo123459** [comment]
Thanks for the contribution! :)

**Pull request merged**